sa/storage-authority.go

```diff
@@ -869,3 +869,30 @@ func (ssa *SQLStorageAuthority) AddSCTReceipt(sct core.SignedCertificateTimestam
 	}
 	return err
 }
+
+// AddNameSet adds the hashed set of names from a certificate to the nameSets
+// table
+func (ssa *SQLStorageAuthority) AddNameSet(nameSet core.NameSet) error {
+	return ssa.dbMap.Insert(&nameSet)
+}
+
+// GetNameSet reutrns the NameSet object with hash |setHash|
+func (ssa *SQLStorageAuthority) GetNameSet(setHash []byte) (*core.NameSet, error) {
+	var nameSet core.NameSet
+	err := ssa.dbMap.SelectOne(&nameSet, "SELECT * FROM nameSets WHERE setHash = ?", setHash)
+	return &nameSet, err
+}
+
+// CountValidNameSets reutrns the number of currently valid sets with hash |setHash|
+func (ssa *SQLStorageAuthority) CountValidNameSets(setHash []byte) (int64, error) {
```

Comment:
I think we'll avoid bugs if we send the names, instead. That way we can change hash functions or how we sort them or whatever in the future.

That also let us avoid having to make a new function in core